### PR TITLE
Document Fix:

### PR DESCRIPTION
**Pull request by Gyx-One**
1.training tutorial refined.

### CODE REVIEW

File: docs/markdown/tutorial/training.md [hunk 1]
```diff
@@ -1,173 +1,437 @@
 # Tutorial for Training Library
+Hyperpose python training library provides a one-step yet flexible platform for developing pose estimation models.
+
+Based on the intented usage, there are two kinds of users of the pose estimation development platform:
```
[the remark this answers is not in the document]
Comment (ganler) [reply docs/markdown/tutorial/training.md:4]: Based on the intended usage, there are two major categories of user requirements regarding pose estimation systems:

File: docs/markdown/tutorial/training.md [hunk 2]
```diff
@@ -1,173 +1,437 @@
...
+- **1.Users aim to adapt the existing algorithms to specific deployment scenario.**
```
[the remark this answers is not in the document]
Comment (ganler) [reply docs/markdown/tutorial/training.md:5]:
```markdown
1. Adapting existing algorithms to specific deployment scenarios.
```

File: docs/markdown/tutorial/training.md [hunk 3]
```diff
@@ -1,173 +1,437 @@
...
+
+  **Example:** Search for the pose estimation model architecture with best accuracy conditioned on the limited hardware resources.
+
+- **2.Users aim to design and polish their own pose estimation algorithms.**
```
Comment (ganler) [inline docs/markdown/tutorial/training.md:9]:
```markdown
Developing customized pose estimation algorithms.
```

File: docs/markdown/tutorial/training.md [hunk 4]
```diff
@@ -1,173 +1,437 @@
...
+
+  **Example:** Explore new pose estimation algorithms with the help of existing dataset generation pipeline and model architectures. 
+
+To meet the requirements of two kinds of users, Hyperpose provides both abundant High-level APIs with integrated pipelines (for first kind of users) and fine-grained APIs with in-depth customisation (for second kidn of users).
```
Comment (ganler) [inline docs/markdown/tutorial/training.md:13]: - "two kinds of users" -> "2 kinds of user requirements"
- "Hyperpose" -> "HyperPose"
- "abundant" -> "rich"
- "High-level" -> "high-level"
- "for second kidn of users" -> "for the second kind of requirement"

File: docs/markdown/tutorial/training.md [hunk 5]
```diff
@@ -1,173 +1,437 @@
...
+
+  **Example:** Search for the pose estimation model architecture with best accuracy conditioned on the limited hardware resources.
+
+- **2.Users aim to design and polish their own pose estimation algorithms.**
+
+  **Example:** Explore new pose estimation algorithms with the help of existing dataset generation pipeline and model architectures. 
+
+To meet the requirements of two kinds of users, Hyperpose provides both abundant High-level APIs with integrated pipelines (for first kind of users) and fine-grained APIs with in-depth customisation (for second kidn of users). 
+
 Up to now, Hyperpose provides:
```
Comment (ganler) [inline docs/markdown/tutorial/training.md:15]: HyperPose

File: docs/markdown/tutorial/training.md [hunk 6]
```diff
@@ -1,173 +1,437 @@
...
-> Mobilenet variants(Dilated Mobilenet,MobilenetThin,MobilenetSmall, located in the preset model architectures)   
-* 2 types of popular dataset
-> COCO
-> MPII
-* extensions
-> user-defined dataset
-> user-defined model architecture
-> pre-processors and post-processors 
+
+- **5 types of preset model algorithms**
```
Comment (ganler) [inline docs/markdown/tutorial/training.md:17]: I think you might also want to split the contents into several paragraphs.

e.g.,

```markdown
## User requirements (? but IMHO this seems to be a bit minor)
```

File: docs/markdown/tutorial/training.md [hunk 7]
```diff
@@ -1,173 +1,437 @@
...
+  - Openpose:
```
Comment (ganler) [inline docs/markdown/tutorial/training.md:18]: Can we have a table here since it is quite structural here?

Comment (ganler) [reply docs/markdown/tutorial/training.md:18]:
```markdown
| Algorithm                                                    | Description |
| ------------------------------------------------------------ | ----------- |
| [OpenPose](https://openaccess.thecvf.com/content_cvpr_2017/html/Cao_Realtime_Multi-Person_2D_CVPR_2017_paper.html) |             |
| Lightweight OpenPose                                         |             |
| Mobilenet ThinOpenPose                                       |             |
| PoseProposal                                                 |             |
| PifPaf                                                       |             |



```

File: docs/markdown/tutorial/training.md [hunk 8]
```diff
@@ -1,173 +1,437 @@
...
+  
+    Original Openpose algorithm. Paper is [here](https://openaccess.thecvf.com/content_cvpr_2017/html/Cao_Realtime_Multi-Person_2D_CVPR_2017_paper.html).
+
+  - LightweightOpenpose:
+
+    A light weight variant of Openpose with optimized prediction branch, featured with fast inference. Paper is [here](https://arxiv.org/abs/1811.12004).
```
Comment (ganler) [inline docs/markdown/tutorial/training.md:24]: “light-weight”

File: docs/markdown/tutorial/training.md [hunk 9]
```diff
@@ -1,173 +1,437 @@
...
+
+  - MobilenetThinOpenpose:
+  
+    A light weight variant of Openpose with adapted Mobilnet backbone, featured with fast inference. Origin repository is [here](https://github.com/jiajunhua/ildoonet-tf-pose-estimation).
```
Comment (ganler) [inline docs/markdown/tutorial/training.md:28]: "light-weight"

File: docs/markdown/tutorial/training.md [hunk 10]
```diff
@@ -1,173 +1,437 @@
...
+
+  - Poseproposal:
+  
+    A pose estimation algorithm that models key point detection as object detection with bounding box, featured with fast inference and post-processing. Paper is [here](https://openaccess.thecvf.com/content_ECCV_2018/html/Sekii_Pose_Proposal_Networks_ECCV_2018_paper.html).
+
+  - Pifpaf:
+
+    An accurate pose esitmation algorithm that generates high resolution confidence map, featured with high accuracy over low resolution images. Paper is [here](https://openaccess.thecvf.com/content_CVPR_2019/html/Kreiss_PifPaf_Composite_Fields_for_Human_Pose_Estimation_CVPR_2019_paper.html).
+
+- **10 types of common model backbone**
```
Comment (ganler) [inline docs/markdown/tutorial/training.md:38]: Would also love to have them tabelized here.

File: docs/markdown/tutorial/training.md [hunk 11]
```diff
@@ -1,173 +1,437 @@
...
+  - Custome dataset
+    - User-add dataset
+
+      User add their self-collected data into preset dataset generation pipeline for train and evaluation.
+
+    - User-defined dataset
+
+      User define their own dataset class to take over the whole dataset generation pipeline.
+
+  - Custome model
```
Comment (ganler) [inline docs/markdown/tutorial/training.md:79]: TYPO ❌ “custome” ✅ “customized”

Comment (ganler) [reply docs/markdown/tutorial/training.md:79]: There's a lot of such typos. Please check with grammarly.

File: docs/markdown/tutorial/training.md [hunk 12]
```diff
@@ -1,173 +1,437 @@
...
+
+  - Custome pipeline
+
+      User use the provided pre-processors, post-processors and visualizers to assemble their own training or evalution pipeline.
+
 
 ## Integrated pipeline
-Hyperpose extract similiar models into a model class. For now, there are two classes: Openpose classes and Poseproposal classes.
-all model architecture can be devided into one of them.  
+Hyperpose provides integrated train, evaluate and test pipeline with various High-level APIs for user to fast adapt the existing pose estimation algorithms for their specific usage scenario.
```
Comment (ganler) [inline docs/markdown/tutorial/training.md:89]: Hyperpose integrates training, evaluation and testing pipeline with various High-level APIs for quickly adapting the existing pose estimation algorithms for their customized usage scenarios.

File: docs/markdown/tutorial/training.md [hunk 13]
```diff
@@ -1,173 +1,437 @@
...
+
+The whole procedure can be devided into two parts:
+
+**In the first part**, users use the **set** APIs of the **Config** module to set up the components of the pipeline. User can set up from the general settings such as algorithm type, network architecture and dataset type, to the detailed settings including training batch size, save interval and learning rate etc.
```
Comment (ganler) [inline docs/markdown/tutorial/training.md:93]: I think in general people use "``" (code blocks) to mark APIs.

File: docs/markdown/tutorial/training.md [hunk 14]
```diff
@@ -1,173 +1,437 @@
...
+
+**In the second part**, users use the **get** APIs of the **Model** module and the **Dataset** module to assemble the system. After the configuration is finished, user could get a *config* object containing all the configuration. Pass the *config* object to the **get** APIs, users get the configured model, dataset, and the train or evaluate pipeline. 
+
+
+The critical **set** APIs are below: (**necessary**)
+
+- **Config.set_model_name**
+ 
+  Recieve a string, which is used as the name for identifing the current model.
```
Comment (ganler) [inline docs/markdown/tutorial/training.md:102]: "receive"
"identifying"

Comment (ganler) [reply docs/markdown/tutorial/training.md:102]: Uniquely identify the model with a name (a string).

**Review by ganler — CHANGES_REQUESTED** [review]
Could you first resolve the issues I mentioned above? Thank you for your contribution.

File: docs/markdown/tutorial/training.md [hunk 15]
```diff
@@ -1,173 +1,437 @@
 # Tutorial for Training Library
+Hyperpose python training library provides a one-step yet flexible platform for developing pose estimation models.
```
[the remark this answers is not in the document]
Comment (Gyx-One) [reply docs/markdown/tutorial/training.md:2]: Thanks for remind :)

File: docs/markdown/tutorial/training.md (hunk 7 above)
Comment (Gyx-One) [reply docs/markdown/tutorial/training.md:18]: Good idea

File: docs/markdown/tutorial/training.md (hunk 13 above)
Comment (Gyx-One) [reply docs/markdown/tutorial/training.md:93]: make sense

**Comment by Gyx-One** [comment]
Thanks a lot for advice! @ganler

**Comment by Gyx-One** [comment]
Table replaced. checking grammarly.

File: docs/markdown/tutorial/training.md (hunk 4 above)
Comment (Gyx-One) [reply docs/markdown/tutorial/training.md]: fixed.

**Comment by ganler** [comment]
@Gyx-One Thanks for the hard work. I think there is still some space for us to make it look more like a concrete tutorial.

To make it more concrete, let's:

- Section 1: first show a step-2-step code example regarding training;
- Section 2: how to do some modifications to achieve other goals (e.g., how to change to other models? how to change to other algorithms? How to enable parallel training?) And please do identify the corresponding APIs (using minimal code example) regarding the feature (otherwise users may still have no idea on how to apply them);
- Other sections...

Also, I'd like to mention some formatting issues regarding your recent PR on documentation.

- If you are writing some API stuff, use "``" to mark them. e.g., `vgg19_backbone`, `Config.DATA.USERDEF`.
    - ✅: one can refer `Model.LightweightOpenpose` class for reference.
    - ❌: one can refer **Model.LightweightOpenpose** class for reference.
- Whenever you finished a paragraph, a title ("###"), add an additional line-break to split the paragragraph.
- Add additional space for some python operators or commas. 
    - ✅: `pd_conf_map, pd_paf_map = my_model.forward(input_image[np.newaxis,...])`
    - ❌: `pd_conf_map,pd_paf_map=my_model.forward(input_image[np.newaxis,...])`

You can refer to my last commit on `yixiao` branch to see how I made changes on your doc.

**Review by ganler — APPROVED** [review]
OK. Let's first merge this one since it has already improved the quality a lot. But remember to follow up. Thanks for the hard work! @Gyx-One

**Pull request merged**